knowledge-base/catch-all-exceptions-globally.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Catch all exceptions globally using ErrorBoundary
+description: How to use ErrorBoundary to catch all exceptions globally?
+type: how-to
+page_title: Catch all exceptions globally using ErrorBoundary
+slug: common-kb-catch-all-exceptions-globally
+position: 
+tags: telerik,blazor,catch,all,exceptions,globally,error,boundary
+ticketid: 
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>UI for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+Is it possible to catch all errors in Blazor Server like in MVC 5 in global.asax on Application_Error?
+
+We are trying to integrate `ErrorBoundary` component but it seems it does not catch exceptions thrown from Window component. What is the correct way of using ErrorBoundary for the Telerik components?
+
+## Solution
+
+As of .NET 6, you can use [`ErrorBoundary` component](https://docs.microsoft.com/en-us/aspnet/core/blazor/fundamentals/handle-errors?view=aspnetcore-6.0#error-boundaries) to catch exceptions.
+
+To do so, wrap the existing content in an `ErrorBoundary` component. It will render error UI when an unhandled exception is thrown. The app continues to function normally, but the error boundary handles the exceptions in the affected component.
+
+`ErrorBoundary` can be wrapped around any Blazor component or around the `@Body` element in the MainLaout.razor, so it catches all exceptions globally.
+
+````CSHTML
+<div class="page">
+    <div class="sidebar">
+        <NavMenu />
+    </div>
+
+    <div class="main">
+        <div class="top-row px-4">
+            <a href="https://docs.microsoft.com/en-us/aspnet/" target="_blank">About</a>
+        </div>
+
+        <div class="content px-4">
+            <ErrorBoundary>
+                @Body
+            </ErrorBoundary>
+        </div>
+    </div>
+</div>
+````
+
+The `ErrorBoundary` component can be used to catch exceptions in the Telerik components as well. While the common approach to catch all exceptions on global level is to wrap it around the `@Body` element, this will not cover errors thrown in components such as Window, Dialog, other popups. These components are not rendered in their place of declaration and essentially they are outside the body.
```

Comment:
```suggestion
The `ErrorBoundary` component can catch exceptions in the Telerik Blazor components as well. The usual approach to catch all exceptions is to wrap `ErrorBoundary` around the `@Body` element. However, this may not catch errors thrown in components such as Window, Dialog, and other popups. These components are not rendered in their place of declaration and essentially they are outside the body.
```